### DISCUSSION TITLE
Spinning Cog of Death

### DISCUSSION THREAD

**Discussion by robotDad414** [homebridge-simplisafe3/homebridge-simplisafe3#348] [General]
I recently started having problems with all of my plugins after the homebridge ui update. I ended up completely uninstalling homebridge via terminal and then reinstalling everything. After I reinstalled I was able to get all of my plugins working properly again with little effort except simplisafe. I'm getting authmanager errors. 

I've uninstalled and reinstalled several times via both the config ui and terminal and I'm still having the same issue. See below for the error log. 

/usr/local/lib/node_modules/homebridge-simplisafe3/lib/authManager.js:141
      const account = JSON.parse(fileContents);
                           ^
SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at SimpliSafe3AuthenticationManager.parseAccountsFile (/usr/local/lib/node_modules/homebridge-simplisafe3/lib/authManager.js:141:28)
    at new SimpliSafe3AuthenticationManager (/usr/local/lib/node_modules/homebridge-simplisafe3/lib/authManager.js:116:10)
    at new SS3Platform (/usr/local/lib/node_modules/homebridge-simplisafe3/index.js:68:24)
    at ChildBridgeFork.startBridge (/usr/local/lib/node_modules/homebridge/src/childBridgeFork.ts:150:42)

**Comment by shamoon** [comment]
I’m guessing your auth file `simplisafe3auth.json` got corrupt in the process, try removing it completely. Location can vary by installation type but is commonly `~/.homebridge`, but storage path can be found in the UI under System Information

**Comment by robotDad414** [comment]
That file didn't exist anywhere, but the issue is resolved. When "completely" removing homebridge it wasn't clearing my cache file.


 I ended up having to run all the commands below to get rid of everything and then start from scratch. 

Commands to completely remove homebridge and start fresh: 

Remove startup service:
sudo hb-service uninstall

Remove Homebridge Config UI:
sudo npm uninstall -g homebridge homebridge-config-ui-x

Remove all node modules associated with homebridge:
sudo rm -rf /usr/local/{lib/node{,/.npm,_modules},bin,share/man}/{npm*,node*,man1/node*}
sudo rm -rf /opt/local/bin/node /opt/local/include/node /opt/local/lib/node_modules
sudo rm -rf /usr/local/lib/node_modules/

Remove cached accessories for current homebridge install:
rm -rf ~/.homebridge/persist/
rm -rf ~/.homebridge/accessories/
